src/transformers/models/blip/modeling_blip.py

```diff
@@ -0,0 +1,1453 @@
+        >>> from PIL import Image
+        >>> import requests
+        >>> from transformers import BlipProcessor, BlipForQuestionAnswering
+
+        >>> model = BlipForQuestionAnswering.from_pretrained("Salesforce/blip-vqa-base")
+        >>> processor = BlipProcessor.from_pretrained("Salesforce/blip-vqa-base")
+
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+
+        >>> inputs = processor(images=image, return_tensors="pt")
+
+        >>> outputs = model.generate(**inputs)
+        ```"""
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+        batch_size, _ = input_ids.shape
+
+        vision_outputs = self.vision_model(
+            pixel_values=pixel_values,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        image_embeds = vision_outputs[0]
+        image_atts = torch.ones(image_embeds.size()[:-1], dtype=torch.long)
+
+        question_embeds = self.text_encoder(
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+            encoder_hidden_states=image_embeds,
+            encoder_attention_mask=image_atts,
+            return_dict=return_dict,
+        )
+
+        question_embeds = question_embeds[0] if not return_dict else question_embeds.last_hidden_state
+
+        if answer_input_ids is None:
+            answer_input_ids = torch.LongTensor([self.decoder_bos_token_id]).repeat((batch_size, 1))
+        answer_targets = answer_input_ids.masked_fill(answer_input_ids == self.decoder_pad_token_id, -100)
+
+        answer_output = self.text_decoder(
+            input_ids=answer_input_ids,
+            attention_mask=answer_attention_mask,
+            encoder_hidden_states=question_embeds,
+            encoder_attention_mask=attention_mask,
+            labels=answer_targets,
+            return_dict=return_dict,
+            reduction="none",
+        )
+
+        decoder_loss = answer_output.loss.mean() if return_dict else answer_output[0].mean()
+
+        if not return_dict:
+            outputs = (decoder_loss, image_embeds, vision_outputs[0]) + vision_outputs[2:]
+            return tuple(output for output in outputs if output is not None)
+
+        return BlipTextVisionModelOutput(
+            loss=decoder_loss,
+            image_embeds=image_embeds,
+            last_hidden_state=vision_outputs.last_hidden_state,
+            hidden_states=vision_outputs.hidden_states,
+            attentions=vision_outputs.attentions,
+        )
+
+    @torch.no_grad()
+    def generate(
+        self,
+        input_ids: torch.LongTensor,
+        pixel_values: torch.FloatTensor,
+        attention_mask: Optional[torch.LongTensor] = None,
+        pixel_mask: Optional[torch.LongTensor] = None,
+        token_type_ids: Optional[torch.LongTensor] = None,
```

Comment:
Also it doesn't seem this model uses token type IDs

Reply:
fixed in [4968a66](https://github.com/huggingface/transformers/pull/20716/commits/4968a662446b834c82ef377991fc988830d54bdc)